RecFCCeeCalorimeter/src/components/AugmentClustersFCCee.cpp

```diff
@@ -175,48 +359,242 @@ StatusCode AugmentClustersFCCee::execute([[maybe_unused]] const EventContext &ev
     {
       if (k > 0)
-      {
         startPositionToFill += m_numLayers[k - 1];
-      }
       int systemID = m_systemIDs[k];
       std::string layerField = m_layerFieldNames[k];
+      std::string thetaField = m_thetaFieldNames[k];
+      std::string moduleField = m_moduleFieldNames[k];
       std::string readoutName = m_readoutNames[k];
       dd4hep::DDSegmentation::BitFieldCoder *decoder = m_geoSvc->getDetector()->readout(readoutName).idSpec().decoder();
 
+      // loop over the cells
       for (auto cell = newCluster.hits_begin(); cell != newCluster.hits_end(); cell++)
       {
         dd4hep::DDSegmentation::CellID cID = cell->getCellID();
         int sysId = decoder->get(cID, "system");
         if (sysId != systemID)
           continue;
         uint layer = decoder->get(cID, layerField);
+        int theta_id = decoder->get(cID, thetaField);
+        int module_id = decoder->get(cID, moduleField);
+
         double eCell = cell->getEnergy();
-        double weightLog = std::max(0., m_thetaRecalcLayerWeights[k][layer] + log(eCell / sumEnLayer[layer]));
+        double weightLog = std::max(0., m_thetaRecalcLayerWeights[k][layer] + log(eCell / sumEnLayer[layer]));  // layer+startPositionToFill ?
         TVector3 v = TVector3(cell->getPosition().x, cell->getPosition().y, cell->getPosition().z);
         double theta = v.Theta();
         double phi = v.Phi();
+
         // for clusters that are around the -pi<->pi transition, we want to avoid averaging
         // over phi values that might differ by 2pi. in that case, for cells with negative
         // phi we add two pi, so that we average phi values all close to pi
-        if (isClusterPhiNearPi && phi < 0.)
+        if (isClusterPhiNearPi && phi < 0.) {
           phi += TMath::TwoPi();
+        }
+        if (systemID == systemID_EMB && isResetModuleID && module_id > nModules[k]/2) {
+          module_id -= nModules[k];  // transition near 1535..0, reset the module ID
+        }
+
         if (m_thetaRecalcLayerWeights[k][layer]<0)
           sumThetaLayer[layer+startPositionToFill] += (eCell * theta);
         else
           sumThetaLayer[layer+startPositionToFill] += (weightLog * theta);
         sumWeightLayer[layer+startPositionToFill] += weightLog;
         sumPhiLayer[layer+startPositionToFill] += (eCell * phi);
-      }
+
+        // do pi0/photon shape var only for EMB
+        if (m_do_pi0_photon_shapeVar && systemID == systemID_EMB) {
+          // E, theta_id, and module_id of cells in layer
+          vec_E_cell_layer[layer+startPositionToFill].push_back(eCell);
+          vec_theta_cell_layer[layer+startPositionToFill].push_back(theta_id);
+          vec_module_cell_layer[layer+startPositionToFill].push_back(module_id);
+          // sum them for width in theta/module calculation
+          theta2_E_layer[layer+startPositionToFill] += theta_id * theta_id * eCell;
+          theta_E_layer[layer+startPositionToFill] += theta_id * eCell;
```

Comment:
Since the theta response is better with logE weights, have you tried to calculate the w_theta from the mean theta and theta2 using the logE weights instead of the Ecell weights?